 __tests__/pet.test.js

```diff
@@ -0,0 +1,247 @@
+const Pet = require("../src/pet");
+
+describe("constructor", () => {
+  it("returns an object", () => {
+    expect(new Pet("Ted")).toBeInstanceOf(Object);
+  });
+});
+
+describe("constructor", () => {
+  it("sets the name property", () => {
+    const pet = new Pet("Ted");
+
+    expect(pet.name).toEqual("Ted");
+  });
+});
+
+describe("constructor", () => {
+  it("has an initial age of 0", () => {
+    const pet = new Pet("Ted");
+
+    expect(pet.age).toEqual(0);
+  });
+});
+
+describe("constructor", () => {
+  it("has an initial hunger of 0", () => {
+    const pet = new Pet("Ted");
+
+    expect(pet.hunger).toEqual(0);
+  });
+});
+
+describe("constructor", () => {
+  it("has an initial fitness of 10", () => {
+    const pet = new Pet("Ted");
+
+    expect(pet.fitness).toEqual(10);
+  });
+});
+
+describe("isAlive", () => {
+  it("checks if the pet is alive", () => {
+    const pet = new Pet("Garfield");
+    pet.isAlive;
+
+    expect(pet.isAlive).toEqual(true);
+  });
+
+  it("returns false if the pets fitness is 0 or less", () => {
+    const pet = new Pet("Garfield");
+
+    pet.fitness = 0;
+    pet.isAlive;
+
+    expect(pet.isAlive).toEqual(false);
+
+    const pet2 = new Pet("Snoopy");
+
+    pet2.fitness = 8;
+    pet2.isAlive;
+
+    expect(pet2.isAlive).toEqual(true);
+  });
+
+  it("returns false if the pets hunger is 10 or more", () => {
+    const pet = new Pet("Garfield");
+
+    pet.hunger = 12;
+    pet.isAlive;
+
+    expect(pet.isAlive).toEqual(false);
+
+    const pet2 = new Pet("Snoopy");
+
+    pet2.hunger = 7;
+    pet2.isAlive;
+
+    expect(pet2.isAlive).toEqual(true);
+  });
+
+  it("returns false if the pets age is 30 or more", () => {
+    const pet = new Pet("Garfield");
+
+    pet.age = 37;
+    pet.isAlive;
+
+    expect(pet.isAlive).toEqual(false);
+
+    const pet2 = new Pet("Snoopy");
+
+    pet2.age = 22;
+    pet2.isAlive;
+
+    expect(pet2.isAlive).toEqual(true);
+  });
```

Comment:
Applying the same ideas as above, this could be re-written as follows:
```
  it("returns false if the pets hunger is 10", () => {
    const pet = new Pet("Garfield");
    pet.hunger = 12;

    expect(pet.isAlive).toEqual(false);
  });

  it("returns false if the pets age is 30", () => {
    const pet = new Pet("Garfield");
    pet.age = 37;

    expect(pet.isAlive).toEqual(false);
  });
```